File: MAP/Lab6/src/View/TextMenu.java
package View;

import Model.Command.*;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class TextMenu
{

    private Map<String, Command> commands;

    public TextMenu() {
        commands = new HashMap<>();
    }

    public void addCommand(Command c){
        commands.put(c.getKey(),c);
    }

    private void printMenu()
    {
        for (Command com : commands.values())
        {
            System.out.println(String.format("%4s : %s", com.getKey(), com.getDescription()));
        }
    }
    public void show()
    {
        Scanner scanner = new Scanner(System.in);
        while(true){
            printMenu();
            System.out.println("Input an option: ");
            Command com = commands.get(scanner.nextLine());
            if (com == null){
                System.out.println("Invalid Option");
                continue;
            }
            com.execute();
        }
    }
}

File: MAP/LAB7-MAP/LAB7-MAP/View/TextMenu.cs
using System;
using System.Collections.Generic;
using MAP_Lab7.Commands;

namespace MAP_Lab7.View
{
    public class TextMenu
    {
        private Dictionary<string, Command> command;

        public TextMenu()
        {
            command = new Dictionary<string, Command>();
        }

        public void AddCommand(Command c)
        {
            command[c.GetKey()] = c;
        }

        private void PrintMenu()
        {
            foreach (Command com in command.Values)
            {
                string line = string.Format(com.GetKey() + " " + com.GetDescription());
                System.Console.WriteLine(line);
            }
        }

        public void Show()
        {

            while (true)
            {
                PrintMenu();
                System.Console.WriteLine("Input an option: ");

                string key = Console.ReadLine();

                Command cmd = command[key];
                if (cmd == null)
                {
                    System.Console.WriteLine("Invalid option !");
                    continue;
                }
                cmd.Execute();
            }
        }
    }
}
